main.go

```diff
@@ -0,0 +1,83 @@
+package main
+
+import (
+	"fmt"
+	"os"
+	"os/signal"
+	"sync"
+
+	"github.com/rs/zerolog/log"
+	component "github.com/th2-net/th2-box-template-go/component"
+	"github.com/th2-net/th2-common-go/schema/factory"
+	rabbitmq "github.com/th2-net/th2-common-go/schema/modules/mqModule"
+	"github.com/th2-net/th2-common-go/schema/queue/message"
+)
+
+func main() {
+
+	var closingFunctions []func()
+	wait := shutdown(&closingFunctions)
+
+	newFactory := factory.NewFactory()
+	if err := newFactory.Register(rabbitmq.NewRabbitMQModule); err != nil {
+		log.Fatal().Err(err).Msg("Registering RabbitMQModule failed")
+	}
+	closingFunctions = append(closingFunctions, func() { newFactory.Close() })
```

Comment:
```suggestion
	closingFunctions = append(closingFunctions, func() { newFactory.Close() })
	if err := newFactory.Register(rabbitmq.NewRabbitMQModule); err != nil {
		log.Fatal().Err(err).Msg("Registering RabbitMQModule failed")
	}
```
register closing function right after object creation because shutdown hook can be call in any time